src/Raven.Server/Utils/BackupUtils.cs

```diff
@@ -616,12 +625,67 @@ await BackupHelper.RunWithRetriesAsync(maxRetries: 10, async () =>
         }
         catch (Exception e)
         {
-            const string message = "Error saving the periodic backup status";
+            string message = $"Error saving the periodic backup status in {nameof(UpdatePeriodicBackupStatusCommand)}";
+            Exception fullException = e;
+            try
+            {
+                // if cluster command failed then we save the status locally on our own
+                serverStore.DatabaseInfoCache.BackupStatusStorage.Insert(status, databaseName);
+                message += $"{Environment.NewLine}Saving the local backup status directly succeeded";
+            }
+            catch (Exception innerException)
+            {
+                message += $"{Environment.NewLine}Attempt at saving the local status directly failed as well";
+                fullException = new AggregateException(e, innerException);
+            }
 
             if (logger.IsOperationsEnabled)
-                logger.Operations(message, e);
+                logger.Operations(message, fullException);
+
+            backupResult?.AddError($"{message}{Environment.NewLine}{fullException}");
+        }
+    }
+
+    internal static readonly DateTime BackupScheduleSearchLowerBound = new(year: 2015, month: 1, day: 1);
 
-            backupResult?.AddError($"{message}{Environment.NewLine}{e}");
+    /// <summary>
+    /// Finds the last occurrence of a cron schedule at or before a given point in time.
+    /// Throws ArgumentOutOfRangeException for dates before 2015-01-01.
+    /// </summary>
+    internal static DateTime? GetLastOccurrence(CrontabSchedule schedule, DateTime baseTime)
+    {
+        if (baseTime < BackupScheduleSearchLowerBound)
+            throw new ArgumentOutOfRangeException(nameof(baseTime), $"Dates before {BackupScheduleSearchLowerBound:yyyy-MM-dd} are not supported for backup calculations.");
+
+        // This solves all exact match cases.
+        if (schedule.GetNextOccurrence(baseTime.AddTicks(-1)) == baseTime)
+            return baseTime;
+
+        // Estimate the schedule's interval to calculate a safe starting point for our backward search.
+        var t1 = schedule.GetNextOccurrence(BackupScheduleSearchLowerBound);
+        var t2 = schedule.GetNextOccurrence(t1);
+        var interval = t2 - t1;
+
+        if (interval < TimeSpan.FromMinutes(1))
+            interval = TimeSpan.FromMinutes(1); // Ensure a minimum jump to prevent excessive iterations
+
+        var searchStart = baseTime.Subtract(interval.Multiply(2));
+
+        if (searchStart < BackupScheduleSearchLowerBound)
+            searchStart = BackupScheduleSearchLowerBound;
+
+        DateTime? lastFound = null;
+        // We start searching from one tick *before* our searchStart to ensure searchStart itself is included if it's a valid time.
+        var searchCursor = searchStart.AddTicks(-1);
+
+        while (true)
```

Comment:
please verify this logic

Reply:
Yes, you're right, my initial solution was very limited and completely unprepared for complex cron expressions.

I delved into the issue and implemented the `CrontabScheduleExtensions` class. 
I also significantly expanded the test suite.